File: backend/crates/pm-core/src/models/sprint_status.rs
use crate::{CoreError, CoreResult};

use std::panic::Location;
use std::str::FromStr;

use error_location::ErrorLocation;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum SprintStatus {
    Planned,
    Active,
    Completed,
    Cancelled,
}

impl SprintStatus {
    pub fn as_str(&self) -> &str {
        match self {
            Self::Planned => "planned",
            Self::Active => "active",
            Self::Completed => "completed",
            Self::Cancelled => "cancelled",
        }
    }
}

impl FromStr for SprintStatus {
    type Err = CoreError;

    #[track_caller]
    fn from_str(s: &str) -> CoreResult<Self> {
        match s {
            "planned" => Ok(Self::Planned),
            "active" => Ok(Self::Active),
            "completed" => Ok(Self::Completed),
            "cancelled" => Ok(Self::Cancelled),
            _ => Err(CoreError::InvalidSprintStatus {
                value: s.to_string(),
                location: ErrorLocation::from(Location::caller()),
            }),
        }
    }
}

File: frontend/ProjectManagement.Core/Models/SprintStatus.cs
namespace ProjectManagement.Core.Models;

public enum SprintStatus                                                                                                                         
{                                                                                                                                                
  Planned = 1,                                                                                                                                 
  Active = 2,                                                                                                                                  
  Completed = 3,                                                                                                                               
  Cancelled = 4                                                                                                                                
}

public static class SprintStatusExtensions
{
    public static string ToDisplayString(this SprintStatus status) => status switch
    {
        SprintStatus.Planned => "Planned",
        SprintStatus.Active => "Active",
        SprintStatus.Completed => "Completed",
        SprintStatus.Cancelled => "Cancelled",
        _ => throw new ArgumentOutOfRangeException(nameof(status))
    };
}